code/game/mecha/mecha_defense.dm

```diff
@@ -266,4 +266,33 @@
 /obj/mecha/mech_melee_attack(obj/mecha/M)
 	if(M.damtype == BRUTE || M.damtype == BURN)
 		add_logs(M.occupant, src, "attacked", M, "(INTENT: [uppertext(M.occupant.a_intent)]) (DAMTYPE: [uppertext(M.damtype)])")
-		. = ..()
\ No newline at end of file
+		. = ..()
+
+/obj/mecha/narsie_act()
+	if(occupant)
+		var/mob/living/L = occupant
+		go_out(TRUE)
```

Comment:
Yes, which is why it checks if(L)
The actual type of the occupant doesn't actually happen to matter, here, only that it exists, because it has the x_act() proc no matter what.